src/wrapper.cc

```diff
@@ -124,11 +125,14 @@ Dart_CObject CallbackArgBuilder(int num, ...) {
   Dart_CObject c_request_data;
   va_list valist;
   va_start(valist, num);
-  void *request_buffer = malloc(sizeof(uint64_t) * num);
+  // Initializing [request_buffer] with 0 will allow us to ensure padding in

```

Comment:
This comment is not correct. You are always writing also the upper 32 bits of the value when interacting with it through `uin64_t*`.

Reply:
Oh yes, C is implicitly casting for us. Totally missed this part too :).